config/puma.rb

```diff
@@ -15,7 +15,7 @@
 
 # Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
-port ENV.fetch("PORT") { 3000 }
+bind "tcp://0.0.0.0:80"
```

Comment:
I think configurable port is a good thing, especially for development. Also for those not yet using Docker to develop, listening on 80 won't work unless running as root. See Rails default here (`rails app:update` will flag this one too):

https://github.com/rails/rails/blob/3769f1eee6a1ec4f6438155d67477db6d192577f/railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt#L48

Reply:
I agree. The change is partly to bind to all interfaces, but I will fix it to bind to 3000 by default, configurable.

Reply:
I believe you can still override Puma through other env variables run or just running the `puma` command directly. It's also probably not good to bind to `0.0.0.0` by default for development  (security). You also probably don't need to worry about binding in the Docker container.

Reply:
Ok, I see that there's an arg to do this. That seems better. Unfortunately in the container it binds to loopback and won't connect unless configured.

Reply:
Try with `RAILS_ENV=production`. From the `rails server --help`:
```
Usage:
  rails server -u [thin/puma/webrick] [options]

Options:
  -e, [--environment=ENVIRONMENT]              # Specifies the environment to run this server under (test/development/production).
  -p, [--port=port]                            # Runs Rails on the specified port - defaults to 3000.
  -b, [--binding=IP]                           # Binds Rails to the specified IP - defaults to 'localhost' in development and '0.0.0.0' in other environments'.
...
```

Reply:
Ah, makes sense